tests/TEST_FRAMEWORK.md

```diff
@@ -0,0 +1,321 @@
+# Test Framework Documentation
```

Comment:
General testing framework information should be moved to the DEVELOPER.md and this file should be deleted.